sqlmesh/core/context.py

```diff
@@ -325,6 +325,13 @@ def __init__(
 
         self.path, self.config = t.cast(t.Tuple[Path, C], next(iter(self.configs.items())))
 
+        # This allows overriding the default dialect's normalization strategy, so for example
+        # one can do `dialect="duckdb,normalization_strategy=lowercase"` and this will be
+        # applied to the DuckDB dialect globally
+        if "normalization_strategy" in str(self.config.dialect):
+            dialect = Dialect.get_or_raise(self.config.dialect)
+            type(dialect).NORMALIZATION_STRATEGY = dialect.normalization_strategy
```

Comment:
So I can't have different normalization strategies for the same dialect? Furthermore it can't be overridden on per model basis? Should this be a top-level config then instead of being a part of a dialect? Feels a little misleading otherwise.

Reply:
> So I can't have different normalization strategies for the same dialect?

That's right - but it seems reasonable to me, because you can only have a single default strategy for a given dialect. This simply makes the `default_dialect` override that default globally for said dialect, but it _can_ be overridden on a model basis, so if you do `dialect duckdb,normalization_strategy=...` in a model meta, then I'm pretty sure that should take precedence over the class constant set here.

Reply:
> So I can't have different normalization strategies for the same dialect?

Ah, I guess you were asking if you can override that strategy per model? If so, my answer is a bit contradictory. You can override it wherever you can specify a dialect basically.

Reply:
sorry i'm confused, can you elaborate?

Reply:
- You can set `dialect: foo,normalization_strategy=...` at the model defaults config level. This will monkey patch its `NORMALIZATION_STRATEGY` [class constant](https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py#L216), so it applies globally.
- You can override this behavior per model by doing e.g. `dialect foo,normalization_strategy=...`

Reply:
I'm still confused though. Wouldn't setting it in a model still override the strategy globally?

Reply:
how would it work if you set the model config to normarizaltion=x. would that get propogated to the engine adapter?

Reply:
> Wouldn't setting it in a model still override the strategy globally?

Nope, because it gets stored as an attributed in the dialect and we ignore the `NORMALIZATION_STRATEGY` constant (see [constructor](https://github.com/tobymao/sqlglot/blob/main/sqlglot/dialects/dialect.py#L513-L519)). So there's no global overriding.

> how would it work if you set the model config to normarizaltion=x. would that get propogated to the engine adapter?

If you do this at the model default config, then the dialect will be patched => the engine adapter's dialect will also have its strategy changed. This was the main motivation behind the patching.

If we didn't patch it, then setting `normalization=..` in the project config would override the strategy of the dialect stored in the config, but not the one [stored in the engine adapter](https://github.com/TobikoData/sqlmesh/blob/main/sqlmesh/core/engine_adapter/base.py#L116), because we don't really pass the config's dialect to the adapters.